src/Twig/EventsPushingStatisticsExtension.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusFacebookPlugin\Twig;
+
+use Setono\SyliusFacebookPlugin\Model\PixelInterface;
+use Setono\SyliusFacebookPlugin\Repository\PixelEventRepositoryInterface;
+use Setono\SyliusFacebookPlugin\Workflow\SendPixelEventWorkflow;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+final class EventsPushingStatisticsExtension extends AbstractExtension
+{
+    private PixelEventRepositoryInterface $pixelEventRepository;
+
+    public function __construct(PixelEventRepositoryInterface $pixelEventRepository)
+    {
+        $this->pixelEventRepository = $pixelEventRepository;
+    }
+
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('setono_facebook_events_pushing_statistics', [$this, 'getEventsPushingStatistics']),
+        ];
+    }
+
+    /**
+     * @return array<string, int>
+     */
+    public function getEventsPushingStatistics(PixelInterface $pixel): array
+    {
+        $result = [];
+        foreach (SendPixelEventWorkflow::getStates() as $state) {
+            $count = $this->pixelEventRepository->getCountByPixelAndState($pixel, $state);
```

Comment:
Have you remembered to create an index with the properties used for this SELECT?

Reply:
Well, statistics will be viewed rarely by admins.
If we would add an additional index - it would be taking some extra time on every insert/update/remove.
So I would vote to not add indexes for this particular case.

Reply:
👍